subsys/net/ip/l2/ieee802154/Kconfig

```diff
@@ -6,6 +6,7 @@
 
 menuconfig NET_L2_IEEE802154
 	bool "Enable IEEE 802.15.4 Radio"
+	select NET_6LO
```

Comment:
15.4 L2 has nothing directly related to ipv6/6lo.
Comparing it to BT IPSS is actually exactly the wrong comparison: BT IPSS does specify 6lo/ipv6 for it to work.
Whereas 15.4 does not.

Instead, modify 6lo's Kconfig and enable it if 15.4 L2 is enabled. It will be a way better place, result will be the same but more logical.